doc/mgmt/Sonic_Config_Mgmt_HLD.md

```diff
@@ -0,0 +1,187 @@
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+Not Applicable
+
+### 1.1.2 Configuration and Management Requirements
+
+Operations:
+1. Save running configuration to default. <br>
+   CLI Command : write memory <br>
+   EXEC Level <br>
+   Write to file /etc/sonic/config_db.json  <br>
+
+2. Copy configuration from file to running and reload. <br>
+   CLI Command : copy file://<*filename*> running-configuration [overwrite] <br>
+   EXEC Level <br>
+   Parameter: <*filename*>: user input : file://etc/sonic/<*filename*\>  <br> &ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp; config can only be loaded from /etc/sonic/ directory <br>
+   Parameter: reload (optional) - flushDB and reload services. <br>
+
+3. Save running configuration to file.  <br>
+   CLI Command: copy running-configuration file://<*filename*\>  <br>
+   EXEC Level  <br>
+   Parameter: <*filename*> -  user input: file://etc/sonic/<*filename*> <br> &ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;&ensp;   config can only be saved in /etc/sonic/ directory
+
+4. Copy startup configuration to running-configuration and reload. <br>
+CLI Command: copy running-configuration startup-configuration  <br>
+EXEC Level  <br>
+
+The click CLI additionally supports 'config load_mgmt' and 'config load_minigraph' commands. These operations are SONiC specific. The new sonic management framework will handle the management interface related configuration through the NBI interface. The current json format of configDB replaces the xml format of minigraph.
+
+### 1.1.3 Scalability Requirements
+Not Applicable
+
+### 1.1.4 Warm Boot Requirements
+Not Applicable
+
+## 1.2 Design Overview
+### 1.2.1 Basic Approach
+The operations are invoked via RPC construct of the yang interface. The sonic management framework callback is defined in the sonic-annotation.yang file. The operations are executed on the host service via the dBus framework. The dbus hostservice will execute the operations by calling the config click cli script.
+
+### 1.2.2 Container
+This feature is contained within the sonic-mgmt-framework container.
+
+### 1.2.3 SAI Overview
+Not Applicable.
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Not Applicable
+
+## 2.2 Functional Description
+Not Applicable
+
+# 3 Design
+## 3.1 Overview
+Not Applicable
+
+## 3.2 DB Changes
+Describe changes to existing DBs or any new DB being added.
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+Not Applicable
+
+## 3.4 SyncD
+Not Applicable
+
+## 3.5 SAI
+Not Applicable
+
+## 3.6 User Interface
+### 3.6.1 Data Models
+
+A new sonic yang (sonic-config-mgmt.yang) provides the interface for configuration and status.
+
+```
+module: sonic-config-mgmt
+
+  rpcs:
+    +---x copy
+       +---w input
+       |  +---w source?        string
```

Comment:
Please check comments given for the corresponding SONIC YANG.

Reply:
This is updated as per yang.